Sources/Purchasing/Configuration.swift

```diff
@@ -227,6 +233,9 @@ import Foundation
         /// ### Related Symbols
         /// - ``StoreKitVersion``
         @objc public func with(storeKitVersion version: StoreKitVersion) -> Builder {
+            if self.observerMode {
+                Logger.warn(Strings.configure.observer_mode_with_storekit_version)
+            }
```

Comment:
This only works if you happen to set the version _after_ setting observer mode. Instead, let's do this check in `build()`.

Reply:
This checks for usages of this method _after_ setting the observer mode, and the warning log in the observer mode method checks for usages if this method _before_ setting the observer mode (as long as the value is different than the default, otherwise we cannot know)

Reply:
See my comment above. I think it's a smell that something is wrong with the API.